build.sh

```diff
@@ -35,67 +72,113 @@ do
+    $QUIET || echo "OK, looked in the ../html directory and found HTML source there..."
+    HTML_SOURCE=../html
+  else
+    if [ -f ./html/source ]; then
+      $QUIET || echo "OK, looked in the html/ subdirectory here and found HTML source..."
+      HTML_SOURCE=./html
+    else
+      # TODO Before giving up, should we maybe also check $HOME/html? Or anywhere else?
+      $QUIET || echo "Didn't find the HTML source on your system..."
+      $QUIET || echo "OK, cloning it..."
+      cloneHtml
+    fi
+  fi
+else
+  $QUIET || echo "HTML_SOURCE environment variable is set to $HTML_SOURCE; looking for HTML source..."
+  if [ -f "$HTML_SOURCE/source" ]; then
+    $QUIET || echo "OK, looked in the $HTML_SOURCE directory and found HTML source there..."
+  else
+    $QUIET || echo "Looked in the $HTML_SOURCE directory but didn't find HTML source there..."
+    $QUIET || echo "OK, cloning it instead..."
+    cloneHtml
+  fi
+fi
+export HTML_SOURCE
+
+[ -d $HTML_CACHE ]  || mkdir -p $HTML_CACHE || exit 1
+[ -d $HTML_TEMP ]   || mkdir -p $HTML_TEMP || exit 1
+[ -d $HTML_OUTPUT ] || mkdir -p $HTML_OUTPUT || exit 1
+
+if [ ! -d $HTML_CACHE/cldr-data ]; then
+  $QUIET || echo "Checking out CLDR (79 MB)..."
+  svn $($VERBOSE && echo "-v") $($QUIET && echo "-q") \
+    checkout http://www.unicode.org/repos/cldr/trunk/common/main/ $HTML_CACHE/cldr-data
 fi
 
-$QUIET || echo "Examining CLDR...";
-if [ "$DO_UPDATE" == true ] && [ "`svn info -r HEAD .cldr-data | grep -i "Last Changed Rev"`" != "`svn info .cldr-data | grep -i "Last Changed Rev"`" -o ! -s cldr.inc ]; then
-  $QUIET || echo "Updating CLDR...";
-  svn $($VERBOSE && echo "-v") up .cldr-data;
-  perl -T .cldr-processor.pl > cldr.inc;
+$QUIET || echo "Examining CLDR (this takes a moment)...";
+if [ "$DO_UPDATE" == true ] && [ "`svn info -r HEAD $HTML_CACHE/cldr-data | grep -i "Last Changed Rev"`" != "`svn info $HTML_CACHE/cldr-data | grep -i "Last Changed Rev"`" -o ! -s $HTML_TEMP/cldr.inc ]; then
+  $QUIET || echo "Updating CLDR..."
+  svn $($QUIET && echo "-q") up $HTML_CACHE/cldr-data;
+  perl -T .cldr-processor.pl > $HTML_TEMP/cldr.inc;
 fi
 
-if [ "$DO_UPDATE" == true ] || [ ! -f unicode.xml ]; then
-  $QUIET || echo "Downloading unicode.xml...";
+if [ "$DO_UPDATE" == true ] || [ ! -f $HTML_CACHE/unicode.xml ]; then
+  $QUIET || echo "Downloading unicode.xml (can take a short time, depending on your bandwith)...";
   wget $($VERBOSE || echo "--quiet") \
-    -N https://www.w3.org/2003/entities/2007xml/unicode.xml
+    -N https://www.w3.org/2003/entities/2007xml/unicode.xml.zip -O $HTML_CACHE/unicode.xml.zip
+  unzip $($VERBOSE && echo "-v" || echo "-qq") $HTML_CACHE/unicode.xml.zip -d $HTML_CACHE
+  rm $HTML_CACHE/unicode.xml.zip
 fi
 
 # XXX should also check if .entity-processor.py, .entity-processor-json.py, and entities-legacy* have changed
-if [ unicode.xml -nt entities-unicode.inc ]; then
+if [ $HTML_CACHE/unicode.xml -nt $HTML_TEMP/entities-unicode.inc ]; then
```

Comment:
Hmm. I think we should be caching several of these, not temping them. What are all the outputs of this step? They should be entities.json, entities-dtd.url, entities.inc, and cldr.inc, right? I think all those file should end up in the cache, not the temp. Then this line can check if `$HTML_CACHE/unicode.xml -nt $HTML_CACHE/entities.inc`.